File: app/src/main/java/com/yamamz/earthquakemonitor/api/ApiServices.java
package com.yamamz.earthquakemonitor.api;

import com.yamamz.earthquakemonitor.model.EarthquakeGeoJSon;

import retrofit2.Call;
import retrofit2.http.GET;

/**
 * Created by AMRI on 8/24/2017.
 */

public interface ApiServices {
    //last day earthquake
    @GET("earthquakes/feed/v1.0/summary/all_day.geojson")
    Call<EarthquakeGeoJSon> getEarthQuakesLastdayAll();
    @GET("earthquakes/feed/v1.0/summary/1.0_day.geojson")
    Call<EarthquakeGeoJSon> getEarthQuakesLastday1_0();
    @GET("earthquakes/feed/v1.0/summary/2.5_day.geojson")
    Call<EarthquakeGeoJSon> getEarthQuakesLastday2_5();
    @GET("earthquakes/feed/v1.0/summary/4.5_day.geojson")
    Call<EarthquakeGeoJSon> getEarthQuakesLastday4_5();
    @GET("earthquakes/feed/v1.0/summary/significant_day.geojson")
    Call<EarthquakeGeoJSon> getEarthQuakesLastdaySig();


    //last hour earthquake
    @GET("earthquakes/feed/v1.0/summary/all_hour.geojson")
    Call<EarthquakeGeoJSon> getEarthQuakesLasthourAll();
    @GET("earthquakes/feed/v1.0/summary/1.0_hour.geojson")
    Call<EarthquakeGeoJSon> getEarthQuakesLasthour1_0();
    @GET("earthquakes/feed/v1.0/summary/2.5_hour.geojson")
    Call<EarthquakeGeoJSon> getEarthQuakesLasthour2_5();
    @GET("earthquakes/feed/v1.0/summary/4.5_hour.geojson")
    Call<EarthquakeGeoJSon> getEarthQuakesLasthour4_5();
    @GET("earthquakes/feed/v1.0/summary/significant_hour.geojson")
    Call<EarthquakeGeoJSon> getEarthQuakesLasthourSig();


    //last hour earthquake
    @GET("earthquakes/feed/v1.0/summary/all_week.geojson")
    Call<EarthquakeGeoJSon> getEarthQuakesLastweekAll();
    @GET("earthquakes/feed/v1.0/summary/1.0_week.geojson")
    Call<EarthquakeGeoJSon> getEarthQuakesLastweek1_0();
    @GET("earthquakes/feed/v1.0/summary/2.5_week.geojson")
    Call<EarthquakeGeoJSon> getEarthQuakesLastweek2_5();
    @GET("earthquakes/feed/v1.0/summary/4.5_week.geojson")
    Call<EarthquakeGeoJSon> getEarthQuakesLastweek4_5();
    @GET("earthquakes/feed/v1.0/summary/significant_week.geojson")
    Call<EarthquakeGeoJSon> getEarthQuakesLastweekSig();


    //last hour earthquake
    @GET("earthquakes/feed/v1.0/summary/all_month.geojson")
    Call<EarthquakeGeoJSon> getEarthQuakesLastmonthAll();
    @GET("earthquakes/feed/v1.0/summary/1.0_month.geojson")
    Call<EarthquakeGeoJSon> getEarthQuakesLastmonth1_0();
    @GET("earthquakes/feed/v1.0/summary/2.5_month.geojson")
    Call<EarthquakeGeoJSon> getEarthQuakesLastmonth2_5();
    @GET("earthquakes/feed/v1.0/summary/4.5_month.geojson")
    Call<EarthquakeGeoJSon> getEarthQuakesLastmonth4_5();
    @GET("earthquakes/feed/v1.0/summary/significant_month.geojson")
    Call<EarthquakeGeoJSon> getEarthQuakesLastmonthSig();






}

File: app/src/main/java/com/yamamz/earthquakemonitor/api/ApiServices.kt
package com.yamamz.earthquakemonitor.api

import com.yamamz.earthquakemonitor.model.EarthquakeGeoJSon

import retrofit2.Call
import retrofit2.http.GET

/**
 * Created by AMRI on 8/24/2017.
 */

interface ApiServices {
    //last day earthquake
    @get:GET("earthquakes/feed/v1.0/summary/all_day.geojson")
    val earthQuakesLastdayAll: Call<EarthquakeGeoJSon>
    @get:GET("earthquakes/feed/v1.0/summary/1.0_day.geojson")
    val earthQuakesLastday1_0: Call<EarthquakeGeoJSon>
    @get:GET("earthquakes/feed/v1.0/summary/2.5_day.geojson")
    val earthQuakesLastday2_5: Call<EarthquakeGeoJSon>
    @get:GET("earthquakes/feed/v1.0/summary/4.5_day.geojson")
    val earthQuakesLastday4_5: Call<EarthquakeGeoJSon>
    @get:GET("earthquakes/feed/v1.0/summary/significant_day.geojson")
    val earthQuakesLastdaySig: Call<EarthquakeGeoJSon>


    //last hour earthquake
    @get:GET("earthquakes/feed/v1.0/summary/all_hour.geojson")
    val earthQuakesLasthourAll: Call<EarthquakeGeoJSon>
    @get:GET("earthquakes/feed/v1.0/summary/1.0_hour.geojson")
    val earthQuakesLasthour1_0: Call<EarthquakeGeoJSon>
    @get:GET("earthquakes/feed/v1.0/summary/2.5_hour.geojson")
    val earthQuakesLasthour2_5: Call<EarthquakeGeoJSon>
    @get:GET("earthquakes/feed/v1.0/summary/4.5_hour.geojson")
    val earthQuakesLasthour4_5: Call<EarthquakeGeoJSon>
    @get:GET("earthquakes/feed/v1.0/summary/significant_hour.geojson")
    val earthQuakesLasthourSig: Call<EarthquakeGeoJSon>


    //last hour earthquake
    @get:GET("earthquakes/feed/v1.0/summary/all_week.geojson")
    val earthQuakesLastweekAll: Call<EarthquakeGeoJSon>
    @get:GET("earthquakes/feed/v1.0/summary/1.0_week.geojson")
    val earthQuakesLastweek1_0: Call<EarthquakeGeoJSon>
    @get:GET("earthquakes/feed/v1.0/summary/2.5_week.geojson")
    val earthQuakesLastweek2_5: Call<EarthquakeGeoJSon>
    @get:GET("earthquakes/feed/v1.0/summary/4.5_week.geojson")
    val earthQuakesLastweek4_5: Call<EarthquakeGeoJSon>
    @get:GET("earthquakes/feed/v1.0/summary/significant_week.geojson")
    val earthQuakesLastweekSig: Call<EarthquakeGeoJSon>


    //last hour earthquake
    @get:GET("earthquakes/feed/v1.0/summary/all_month.geojson")
    val earthQuakesLastmonthAll: Call<EarthquakeGeoJSon>
    @get:GET("earthquakes/feed/v1.0/summary/1.0_month.geojson")
    val earthQuakesLastmonth1_0: Call<EarthquakeGeoJSon>
    @get:GET("earthquakes/feed/v1.0/summary/2.5_month.geojson")
    val earthQuakesLastmonth2_5: Call<EarthquakeGeoJSon>
    @get:GET("earthquakes/feed/v1.0/summary/4.5_month.geojson")
    val earthQuakesLastmonth4_5: Call<EarthquakeGeoJSon>
    @get:GET("earthquakes/feed/v1.0/summary/significant_month.geojson")
    val earthQuakesLastmonthSig: Call<EarthquakeGeoJSon>


}
